tests/qasm_test.py

```diff
@@ -15,14 +15,37 @@
 from utilities import run_qir_gen_and_check  # type: ignore
 
 from pytket.qasm import circuit_from_qasm_str
+from pytket.qir.conversion.api import QIRProfile
 
 
 def test_pytket_qir_qasm() -> None:
-    with open("qec.qasm") as my_file:
+    with open("qasm/qec.qasm") as my_file:
         circ = circuit_from_qasm_str(my_file.read())
 
     run_qir_gen_and_check(circ, "test_pytket_qir_qasm")
 
 
+def test_pytket_qir_qasm_classical_0() -> None:
+    with open("qasm/test0.qasm") as my_file:
+        circ = circuit_from_qasm_str(my_file.read(), use_clexpr=True)
+
+    run_qir_gen_and_check(
+        circ,
+        "test_pytket_qir_qasm_classical_0",
+        profile=QIRProfile.PYTKET,
```

Comment:
Is there a reason why you are only checking the pytket profile in this tests?

Reply:
When I use the `ADAPTIVE` profile most of the classical logic is optimized away so it's harder to check the result. (I'm not sure why this isn't the case with the `PYTKET` profile.) But I can add tests using the `ADAPTIVE` profile as well.

Reply:
Pyqir does not know how to interpret the classical functions we use for the registers, so it can't optimise them. I think it is good to have the optimised result, too.